streams/integration-tests/src/test/java/org/apache/kafka/streams/integration/KafkaStreamsTelemetryIntegrationTest.java

```diff
@@ -144,6 +157,51 @@ public void tearDown() throws Exception {
         if (!streamsSecondApplicationProperties.isEmpty()) {
             IntegrationTestUtils.purgeLocalStreamsState(streamsSecondApplicationProperties);
         }
+        if (globalStoreIterator != null) {
+            globalStoreIterator.close();
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"INFO", "DEBUG", "TRACE"})
+    public void shouldPushGlobalThreadMetricsToBroker(final String recordingLevel) throws Exception {
+        streamsApplicationProperties = props(true);
+        streamsApplicationProperties.put(StreamsConfig.METRICS_RECORDING_LEVEL_CONFIG, recordingLevel);
+        IntegrationTestUtils.produceKeyValuesSynchronously(globalStoreTopic,
```

Comment:
Why do we need to produce input data? Seems it's to trigger `process()` to open the iterator? Wondering if we should avoid this, and creating the iterator in `init()` instead?

nit: move `globalStoreTopic` to it's own line to make it easier for the eye to read the code